examples/echo-nats/echo.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"github.com/nats-io/go-nats"
+	"regexp"
+	"github.com/rs/zerolog/log"
+	"strings"
+	"fmt"
+	"time"
+)
+
+const (
+	sub = "slack.event.message"
+	pub = "slack.chat.postMessage"
+	group = "nats-echo-queue"
+)
+
+type NatsClient interface {
+	QueueSubscribe(subject, queue string, cb nats.Handler) (*nats.Subscription, error)
+	Request(subject string, v interface{}, vPtr interface{}, timeout time.Duration) error
+}
+
+func HandleRequest(n NatsClient) {
+	n.QueueSubscribe(sub, group, echoHandler(n))
+}
+
+func echoHandler(n NatsClient) func (m *slackMsg) {
+	return func(m *slackMsg) {
+		if m.validateCommand() {
+
+			msg := slackMsg{
+				Channel:m.Channel,
+				Text: fmt.Sprintf("You said:\n> %s", splitText(m.Text)),
+				ThreadTs:m.Ts,
+			}
+			resp := struct {
+				Ok bool `json:"ok"`
+				Error string `json:"error"`
+			}{}
+			if err := n.Request(pub, &msg, &resp, 10 * time.Second); err != nil {
```

Comment:
You need to check 2 separate error conditions here. The first (the one you are checking) is NATS erroring e.g. set the timeout to  `1*time.Millisecond` and you'll trigger an error). The 2nd is slack returning an error response (which you're not checking) e.g. hardcode the response channel to something nonsense and the flow will fail to echo and you won't get anything logged. Something like this will probably do for this:

```
if err := n.Request(pub, &msg, &resp, 2*time.Second); err != nil || resp.Error != "" {
				log.Error().
					Err(err).
					Str("resp", fmt.Sprintf("%+v", resp)).
					Str("subject", pub).
					Str("message", m.Text).
					Msg("error publishing message to NATS")
			}
```